lib/conncache.h

```diff
@@ -40,6 +40,7 @@ struct conncache {
   struct Curl_hash hash;
   size_t num_conn;
   long next_connection_id;
+  long next_easy_id;
```

Comment:
Both sound fine to me. With a 64 bit type, we reach uniqueness for all practical purposes, I guess (inside the same multi/conn cache).

The next question is then if we want to add this to all the `info()` statements or make this available to the logging functions. For the latter, it would sneak into the API and I am not sure that this is a good idea. But I do not know if someone is attached to the current log strings either.

Reply:
So, some more thinking between issue solving. A backwards compatible way to introduce this:

* go `curl_off_t` for transfer and connection id (do we have printf format for that one btw?)
* add `CURLINFO_EASY_ID` and `CURLINFO_CONN_ID` for the app to retrieve identifiers (-1 for unassigned, e.g. easy not part of multi yet or connection not assigned)
* add `--trace-id` to `curl` that adds these to its verbose/trace output

Do we provide a standard implementation for apps to set as `CURLOPT_DEBUGFUNCTION`? Something along the lines of:

```
CURL_LOGFMT_DEFAULT   (0)
CURL_LOGFMT_IDS    (1 << 0)
CURL_LOGFMT_TIME    (1 << 1)

/* set DEBUGFUNCTION/DEBUGDATA to log to FILE for this easy handle */
curl_easy_set_log(CURL*, FILE*, int fmt)
/* set DEBUGFUNCTION/DEBUGDATA to log to FILE for all easy handles in multi
    that have no other setting already */
curl_multi_set_log(CURLM*, FILE*, int fmt)

```